### PR TITLE
Adding Functionality for Service Principal Authentication

### PR DESCRIPTION
**Pull request by MrAlexPeacock**
Sorry first PR had an issue in the README

### CODE REVIEW

File: azure_databricks_api/__sp_auth.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    def __init__(self, **kwargs):
+        self._tenant_id = kwargs.pop('tenant_id')
+        self._client_id = kwargs.pop('client_id')
+        self._client_secret = kwargs.pop('client_secret')
+        self._databricks_resource_id = kwargs.pop('databricks_resource_id')
+
+    def authenticate(self):
+        auth_headers = {'Content-Type': "application/x-www-form-urlencoded"}
+        application_id = '2ff814a6-3304-4ab8-85cb-cd0e6f879c1d'
```
Comment (ezwiefel) [inline azure_databricks_api/__sp_auth.py:14]: Could you promote this to a class attribute and include a comment (with link) on where the application_id came from?

**Review by ezwiefel — CHANGES_REQUESTED** [review]
@MrAlexPeacock - Apologies on the delay in reviewing this. I'm running a few tests now on my end, but one comment that I wanted to send along while I do that.

Thanks for contributing! Overall, this looks really good.

**Comment by MrAlexPeacock** [comment]
The right way to do this is probably msal. But there's a trade-off to making that a dependency. What are your thoughts there?

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by ezwiefel** [comment]
I agree that it would be the way to get. I definitely hear you on taking that dependency, but in this case, I think it makes sense to use `msal`

**Comment by MrAlexPeacock** [comment]
Sounds good. I’ll put that together in the next couple of days.

**Comment by ezwiefel** [comment]
@MrAlexPeacock - I renamed the `master` branch to `main`... and apparently it had the effect of closing this PR. That's the only reason it was closed. 

If you look at the `main` branch, I've also added some testing and CI/CD capabilities.

**Pull request closed without merging**